File: 2264_largest-3-same-digit-number-in-string/largest-3-same-digit-number-in-string.c
char* largestGoodInteger(char* num) {
    char candidate='\0';
    static char result[4];
    result[3]='\0';
    for(int i=0;i<strlen(num)-2;i++){
        if(num[i]==num[i+1]&&num[i+1]==num[i+2]){
            candidate=num[i]>candidate?num[i]:candidate;
        }
    }
    for(int i=0;i<3;i++){
        result[i]=candidate;
    }
    return result;
}

File: 2264_largest-3-same-digit-number-in-string/largest-3-same-digit-number-in-string.java
class Solution {
    public String largestGoodInteger(String num) {
        int len = num.length();
        int result = -1;
        for (int i = 1; i < len - 1; i++){
            if(num.charAt(i-1) == num.charAt(i) && num.charAt(i) == num.charAt(i+1)){
                result = (num.charAt(i) - '0') > result ? (num.charAt(i) - '0') : result;
            }
        }
        return result == -1 ? "" : Integer.toString(result)+Integer.toString(result)+Integer.toString(result);
    }
}